packages/react-dom/src/__tests__/refs-test.js

```diff
@@ -422,17 +472,20 @@ describe('root level refs', () => {
 
     inst = null;
     ref = jest.fn(value => (inst = value));
-    result = ReactDOM.render(<Comp ref={ref} />, container);
+    root = ReactDOMClient.createRoot(container);
+    await act(() => {
+      root.render(<Comp ref={ref} />);
+    });
 
     expect(ref).toHaveBeenCalledTimes(1);
     expect(inst).toBeInstanceOf(Comp);
-    expect(result).toBe(inst);
```

Comment:
We can't test this with the new API since `root.render` does not return the instance so I copied the original test into `refsLegacy-test.js`.

Reply:
you can get the instance by doing `instance = this` in cDM, not great but we're doing it a lot elsewhere.

Reply:
And that would test `cDM` matches `ref`. 

But not what `ReactDOM.render` returns. `result` wasn't about getting the instance. We already have `inst` which is set from the `ref` spy. `result` was used here to check that `expect(ReactDOM.render(<div ref={current => (instance = current)})).toBe(instance)`.